pkg/process/net/uds_test.go

```diff
@@ -72,3 +77,106 @@ func TestNewUDSListener(t *testing.T) {
 		testWorkingNewUDSListener(tt, dir+"/net.sock")
 	})
 }
+
+type fakeHandler struct {
+	t       *testing.T
+	request string
+}
+
+func (f *fakeHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	f.request = req.URL.String()
+	w.WriteHeader(200)
+}
+
+func testHttpServe(t *testing.T, shouldFailed bool, f *fakeHandler, prefixCmd []string, uid int, gid int) (err error) {
+	dir := t.TempDir()
+	err = os.Chmod(dir, 0777)
+	require.NoError(t, err)
+	err = os.Chmod(dir+"/..", 0777)
+	require.NoError(t, err)
+
+	socketPath := dir + "/test.http.sock"
+	conn, err := net.Listen("unix", socketPath)
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = conn.Close() })
+	err = os.Chmod(socketPath, 0666)
+	require.NoError(t, err)
+
+	go func() {
+		cmd := append(prefixCmd, "curl", "-s", "--unix-socket", socketPath, "http://unix/test")
+		o, err := exec.Command(cmd[0], cmd[1:]...).CombinedOutput()
+		conn.Close() // closing the server
+		if !shouldFailed {
+			if err != nil {
+				t.Log(cmd, string(o))
+			}
+			require.NoError(t, err)
+		} else {
+			require.Error(t, err)
+		}
+	}()
+
+	return HttpServe(conn, f, uid, gid)
+}
+
+func lookupUser(t *testing.T, name string) (usrID int, grpID int, usrIDstr string, grpIDstr string) {
+	usr, err := user.Lookup(name)
+	require.NoError(t, err)
+
+	usrID, err = strconv.Atoi(usr.Uid)
+	require.NoError(t, err)
+
+	grpID, err = strconv.Atoi(usr.Gid)
+	require.NoError(t, err)
+
+	grp, err := user.LookupGroupId(usr.Gid)
+	require.NoError(t, err)
+
+	return usrID, grpID, usr.Username, grp.Name
+}
+
+func checkIfSudoExistAndNotInteractive(t *testing.T) {
+	sudo := exec.Command("/tmp/s.sh", "id")
+	err := sudo.Start()
+	if err != nil {
+		t.Skipf("sudo is not installed %s", err)
+	}
+
+	time.Sleep(100 * time.Millisecond)
+	if _, err := os.Stat(fmt.Sprintf("/proc/%d/exe", sudo.Process.Pid)); err == nil {
+		t.Skipf("sudo is interactive")
+	}
```

Comment:
If the goal is to check whether `sudo` waits for the user to enter a password it seems rather flaky, maybe running `sudo true` with input closed and seeing whether it succeeds might be more reliable ?

Reply:
exec.CommandContext() is the way to go

Reply:
What I meant is that if you run this on a very slow/busy machine, the command could just be preempted and meet the timeout without even running. Even a simple `sudo id` implies fork, exec and probably other syscalls.

Telling `sudo` not to be interactive and checking whether it fails is a more reliable way (and there is no extra wait time):
```go
cmd := exec.Command("sudo", "--non-interactive", "true")
ret := cmd.Run()
if ret != nil {
	t.Skipf("sudo is interactive")
}
```

But I acknowledge that the probability to meet this is very low so do as you wish.